papers/unraveling-recursion/compiling-to-fomega.tex

```diff
@@ -0,0 +1,733 @@
+\input{preamble-lncs}
+
+\title{Unraveling recursion: compiling an IR with recursion to System F}
+
+\author{
+ Roman Kireev\inst{1}\orcidID{0000-0003-4687-2739}
+\and Chad Nester\inst{2}
+\and Michael Peyton Jones\inst{1}\orcidID{0000-0003-0602-1657}
+\and Philip Wadler\inst{2}\orcidID{0000-0001-7619-6378}
+\and Vasilis Gkoumas\inst{1}
+\and Kenneth MacKenzie\inst{1}
+}
+
+\institute{IOHK\\
+\email{\{roman.kireev,michael.peyton-jones,vasilis.gkoumas,kenneth.mackenzie\}@iohk.io}
+\and University of Edinburgh\\
+\email{\{cnester,wadler\}@inf.ed.ac.uk}
+}
+
+
+\usepackage[T1]{fontenc}
+\usepackage[utf8]{inputenc}
+% Add stuff here as latex complains about it
+\DeclareUnicodeCharacter{2080}{\ensuremath{{}_0}}
+\DeclareUnicodeCharacter{2081}{\ensuremath{{}_1}}
+\DeclareUnicodeCharacter{2082}{\ensuremath{{}_2}}
+\DeclareUnicodeCharacter{22A4}{\ensuremath{\top}}
+\DeclareUnicodeCharacter{228E}{\ensuremath{\uplus}}
+\DeclareUnicodeCharacter{1D57}{\ensuremath{{}^t}}
+\DeclareUnicodeCharacter{1D49}{\ensuremath{{}^e}}
+\DeclareUnicodeCharacter{2200}{\ensuremath{\forall}}
+\DeclareUnicodeCharacter{2218}{\ensuremath{\circ}}
+\DeclareUnicodeCharacter{3BB}{\ensuremath{\lambda}}
+\DeclareUnicodeCharacter{2217}{\ensuremath{\ast}}
+\DeclareUnicodeCharacter{2192}{\ensuremath{\rightarrow}}
+\DeclareUnicodeCharacter{00D7}{\ensuremath{\times}}
+
+\usepackage{amsmath}
+\usepackage{amssymb}
+
+\usepackage{verbatim}
+
+\usepackage{etoolbox}
+
+\usepackage{todonotes}
+
+\usepackage{semantic}
+
+\usepackage{mathpartir}
+
+\usepackage{hyperref}
+\usepackage[capitalise, noabbrev]{cleveref}
+
+% If true, we try and include the agda style file and generated latex from
+% the 'latex' dir. If false we just include the original lagda files and
+% redefine 'code' to be a verbatim environment, which is nice for development.
+\newtoggle{lagda}
+\togglefalse{lagda}
+% This is a bit of a hack to make it easy to toggle on including
+% the generated literate Agda latex if it's been built:
+% - Run agda --latex over all the files
+% - echo '\toggletrue{lagda}' > agdaswitch.tex
+% - Built the latex
+\InputIfFileExists{agdaswitch}{}{}
+
+\iftoggle{lagda}{
```

Comment:
I don't remember the details, but wasn't this required to overcome some deficiency in an old version of Agda?  Can we do away with this now?  I think there was a problem with unexpected bits of Agda appearing in the PDF if you used the wrong value of the toggle.

Reply:
Nah, maybe that's irrelevant.  I think that the problem before was that `\begin{code}[hide]` didn't work with the version of Agda style file that we were using. Perhaps it does work now.  That won't remove the need for the `\iftoggle` though, it'll just make it work properly.

Reply:
No, it's there so you can have fast turnaround during development by not having Agda in the loop.

Reply:
Yes, I remembered that later.  There was definitely something broken though (see my second comment), and that may well have gone away now.